File: app/src/main/java/com/robillo/readrush/data/db/model/search/SearchNameRepository.java
package com.robillo.readrush.data.db.model.search;

import android.annotation.SuppressLint;
import android.arch.lifecycle.LiveData;
import android.os.AsyncTask;

import java.util.List;

import javax.inject.Inject;

public class SearchNameRepository {

    @SuppressWarnings("FieldCanBeLocal")
    private final SearchNameDao searchNameDao;

    @Inject
    public SearchNameRepository(SearchNameDao searchNameDao){
        this.searchNameDao = searchNameDao;
    }

    public LiveData<List<SearchName>> getAllSearches() {
        return searchNameDao.getAllSearchNames();
    }

    @SuppressLint("StaticFieldLeak")
    public void insertSearchItem(final SearchName item) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                searchNameDao.insertSearchItem(item);
                return null;
            }
        }.execute();
    }

    @SuppressLint("StaticFieldLeak")
    @SuppressWarnings("unused")
    public void deleteAllSearchItems() {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                searchNameDao.deleteAllSearchNames();
                return null;
            }
        }.execute();
    }

}

File: app/src/main/java/com/robillo/readrush/data/db/model/search/SearchNameRepository.kt
package com.robillo.readrush.data.db.model.search

import android.annotation.SuppressLint
import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MediatorLiveData
import android.os.AsyncTask

import javax.inject.Inject

/**
 * Created by robinkamboj on 30/11/17.
 */

class SearchNameRepository @Inject
constructor(private val searchNameDao: SearchNameDao) {

    val allSearches: LiveData<List<SearchName>>
        get() = searchNameDao.allSearchNames

    @SuppressLint("StaticFieldLeak")
    fun insertSearchItem(item: SearchName) {
        object : AsyncTask<Void, Void, Void>() {
            override fun doInBackground(vararg voids: Void): Void? {
                searchNameDao.insertSearchItem(item)
                return null
            }
        }.execute()
    }

    @Suppress("unused")
    fun deleteAllSearchItems() {
        searchNameDao.deleteAllSearchNames()
    }

}
